include/random_walks/gaussian_cdhr_walk.hpp

```diff
@@ -0,0 +1,153 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#ifndef RANDOM_WALKS_GAUSSIAN_CDHR_WALK_HPP
+#define RANDOM_WALKS_GAUSSIAN_CDHR_WALK_HPP
+
+#include "generators/boost_random_number_generator.hpp"
+
+// Pick a point from the distribution exp(-a_i||x||^2) on the coordinate chord
+template
+<
+    typename NT,
+    typename RandomNumberGenerator
+>
+NT chord_random_point_generator_exp_coord(const NT &l,
+                                          const NT &u,
+                                          const NT &a_i,
+                                          RandomNumberGenerator& rng)
+{
+    NT r, r_val, fn, dis;
+    const NT tol = 0.00000001;
```

Comment:
I think that it is better to set tol outside the function. It is a dirty hack as it is :-)